app/models/ext_management_system.rb

```diff
@@ -707,24 +725,26 @@ def wait_for_ems_workers_removal
     end
   end
 
-  def destroy(task_id = nil)
-    disable!(:validate => false) if enabled?
+  def orchestrate_destroy(task_id = nil)
+    # If the provider hasn't been disabled yet by the high-priority pause! queue method
+    # requeue the destroy operation to be run later.
+    return self.class._queue_task('orchestrate_destroy', [id], task_id, :deliver_on => 1.minute.from_now.utc) if enabled?
```

Comment:
Wonder if it's better to just call `.pause!` here directly.  I think it's idempotent.  If they happen to collide, I would hope that `.pause!` can handle that since it's writing the same value into the column regardless.

Also, I just wanted to use the word idempotent.

Reply:
Though this may create weird races where the orchestrate_destroy destroys the record before the `pause!` queue item updates it?  We might have to be careful with that, so maybe it's _not_ worth it.

Reply:
Yeah the concern was if `pause!` was run by the priority worker at the same time as the `destroy` was run by the generic worker at the same time, https://github.com/ManageIQ/manageiq/pull/21276#discussion_r651350618